### PR TITLE
[CARBONDATA-521]Depends on more stable class of spark in spark2

### PR DESCRIPTION
**Pull request by scwf**
Hot fix for spark2.1, in spark 2.1 the datasource scan node and logical relation changed

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/88/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/89/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/94/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/95/

**Comment by jackylk** [comment]
Add to whitelist

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDictionaryDecoder.scala [hunk 1]
```diff
@@ -223,10 +223,11 @@ case class CarbonDictionaryDecoder(
...
-class CarbonDecoderRDD(relations: Seq[CarbonDecoderRelation],
-                profile: CarbonProfile,
-                aliasMap: CarbonAliasDecoderRelation,
-                prev: RDD[Row],
+class CarbonDecoderRDD(
+    relations: Seq[CarbonDecoderRelation],
+    profile: CarbonProfile,
+    aliasMap: CarbonAliasDecoderRelation,
+    prev: RDD[Row],
                        output: Seq[Attribute])
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDictionaryDecoder.scala]: incorrect indentation

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala [hunk 2]
```diff
@@ -101,12 +103,11 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
...
-                                    projects: Seq[NamedExpression],
-                                    filterPredicates: Seq[Expression],
-                                    scanBuilder: (Seq[Attribute], Array[Filter],
-                                        ArrayBuffer[AttributeReference]) =>
-                                        RDD[InternalRow]) = {
+      relation: LogicalRelation,
+      projects: Seq[NamedExpression],
+      filterPredicates: Seq[Expression],
+      scanBuilder: (Seq[Attribute], Array[Filter],
+      ArrayBuffer[AttributeReference]) => RDD[InternalRow]) = {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala]: incorrect indentation

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala [hunk 3]
```diff
@@ -212,20 +212,20 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
         }
         attr
       }
-      val scan = execution.DataSourceScanExec.create(
+      val scan = new execution.RowDataSourceScanExec(
         updateProject,
         scanBuilder(updateRequestedColumns, candidatePredicates, pushedFilters, needDecoder),
-        relation.relation, metadata, relation.metastoreTableIdentifier)
+        relation.relation, UnknownPartitioning(9), metadata, None)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala]: why is it 9? give comment for this.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala [hunk 4]
```diff
@@ -212,20 +212,20 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
...
       // Don't request columns that are only referenced by pushed filters.
       val requestedColumns =
       (projectSet ++ filterSet -- handledSet).map(relation.attributeMap).toSeq
       val updateRequestedColumns = updateRequestedColumnsFunc(requestedColumns, table, needDecoder)
-      val scan = execution.DataSourceScanExec.create(
+      val scan = new execution.RowDataSourceScanExec(
         updateRequestedColumns,
         scanBuilder(updateRequestedColumns, candidatePredicates, pushedFilters, needDecoder),
-        relation.relation, metadata, relation.metastoreTableIdentifier)
+        relation.relation, UnknownPartitioning(0), metadata, None)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala]: give comment for 0

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/99/

**Pull request closed without merging**